### DISCUSSION TITLE
PyInstaller not working with Multiprocessing

### DISCUSSION THREAD

**Discussion by arvindmn01** [pyinstaller/pyinstaller#7798] [PyInstaller]
I have python application where I am using VOSK and OpenAI whisper model for transcribing audio files and I am also using billiard and multiprocessing python packages for multithreading and multiprocessing. 
this is the project folder structure I want to make the exe file of main_script.

```
C:.
|   audio_processing.py
|   main_script.py  
|   voice_commands.py  
|  
+---assets
|       active.png
|       default.png
|       inactive.png
|
\---whisper-tiny-ct2
        config.json
        model.bin
        tokenizer.json
        vocabulary.txt

```
I was able to generate the exe file. but it is not working properly. it is not taking the audio input. I also tried `multiprocessing.freeze_support()` for enabling the multiprocessing but still getting the same result. 
any help would be appreciated

**Comment by rokm** [comment]
What PyInstaller command are you using?

Are you collecting the files from `assets` and `whisper-tiny-ct2` into your frozen application?

Are you sure this is `multiprcocessing` issue? Does it work if you make a sample program with `vosk` that does not use multiprocessing?

PyInstaller supports only `multiprocessing` - it will likely not work with `billiard`, which appears to be a custom fork of `multiprocessing`...

**Reply by arvindmn01** [reply]
I am using this pyinstaller command `pyinstaller --noconfirm --onefile --console --add-data "audio_processing.py;." --add-data "voice_commands.py;." --add-data "assets;assets/" --collect-binaries "vosk"  "main_script.py"` I am not loading the whisper-tiny-ct2 model for now as i am using the whisper model from hugging face hub. 
this application works fine if I run it on my system. 
and for multiprocessing I am using `billiard`.

**Reply by rokm** [reply]
>  and for multiprocessing I am using `billiard`.

PyInstaller has proper run-time hook only for the stdlib `multiprocessing` module. If you want to use some other multiprocessing module (such as `billiard`), you're on your own.

**Reply by arvindmn01** [reply]
ok, I have replaced all the `billiard` functionalities with `multiprocessing` but still it is not working.
I tried `freeze_support()` at multiple places such as just after importing 
`from multiprocessing import Process, Queue, freeze_support
freeze_support()`
also tried it inside the main function 
```
if __name__=='__main__':
    app = wx.App()
    frame = MyFrame()
    frame.Show()
    freeze_support()
    def on_stop_global():
        frame.on_stop()

    app.MainLoop()

```
still exe file is not working properly

**Reply by rokm** [reply]
Like I asked earlier - if you create a simpler version of your application without any multiprocessing and freeze that using PyInstaller, does it work as expected or not? Because you might actually be having problems with some other component...

Other than that, we cannot really help you debug your application without seeing its code.

**Reply by rokm** [reply]
As a side node, `multiprocessing.freeze_support` should be called before you instantiate your `app` and `frame`; otherwise, this instantiation will happen in every worker subprocess, which is probably not what you want...
